tests/setup_nodes.py

```diff
@@ -579,4 +633,10 @@ async def num_connections():
             node_api_1.full_node.server,
         )
 
+        if connect_to_daemon:
```

Comment:
the `connect_to_daemon` option is a bit tricky. this function (`setup_full_system`) did not previously *start* the daemon, but when `connect_to_daemon` was `True`, it would assume there was one running (on the default port) and connect to it.

Since we can't rely on default ports, this function now *starts* the daemon for `b_tools` (not `b_tools_1`) as well as connects to it. that's why the returned tuple has different sizes on these two cases.